nixos/modules/services/web-apps/peertube.nix

```diff
@@ -0,0 +1,387 @@
+{ lib, pkgs, config, ... }:
+
+let
+  cfg = config.services.peertube;
+
+  configFile = pkgs.writeText  "production.yaml" ''
+    listen:
+      port: ${toString cfg.listenHttp}
+
+    webserver:
+      https: ${toString (if cfg.enableWebHttps then "true" else "false")}
+      hostname: '${cfg.localDomain}'
+      port: ${toString cfg.listenWeb}
+
+    database:
+      hostname: '${cfg.database.host}'
+      port: ${toString cfg.database.port}
+      name: '${cfg.database.name}'
+      username: '${cfg.database.user}'
+
+    redis:
+      hostname: ${toString cfg.redis.host}
+      port: ${toString cfg.redis.port}
+      ${lib.optionalString cfg.redis.enableUnixSocket "socket: '/run/redis/redis.sock'"}
+
+    storage:
+      tmp: '/var/lib/peertube/storage/tmp/'
+      avatars: '/var/lib/peertube/storage/avatars/'
+      videos: '/var/lib/peertube/storage/videos/'
+      streaming_playlists: '/var/lib/peertube/storage/streaming-playlists/'
+      redundancy: '/var/lib/peertube/storage/redundancy/'
+      logs: '/var/lib/peertube/storage/logs/'
+      previews: '/var/lib/peertube/storage/previews/'
+      thumbnails: '/var/lib/peertube/storage/thumbnails/'
+      torrents: '/var/lib/peertube/storage/torrents/'
+      captions: '/var/lib/peertube/storage/captions/'
+      cache: '/var/lib/peertube/storage/cache/'
+      plugins: '/var/lib/peertube/storage/plugins/'
+      client_overrides: '/var/lib/peertube/storage/client-overrides/'
```

Comment:
Can this be overwritable? I have some of those in another drive than others. (Some of them are more like caches and I have them on a faster less redundant drive than the actual video files.)

Reply:
Can they be overwritten in an extra config file?

Reply:
Example:
```
dataDir = "/opt/data";
extraConfig = ''
  storage:
    tmp: '/opt/data/peertube/storage/tmp/'
    logs: '/opt/data/peertube/storage/logs/'
    cache: '/opt/data/peertube/storage/cache/'
'';
```